### PR TITLE
Feat error handling

### PR DESCRIPTION
**Pull request by RomainGuarinoni**
# What's new

## Error handling

We introduced an error handler middleware so that any `500` error will be handle by this middleware. 

Sadly, Express 4.x don't automatticaly call the `next()` function in an `async` controller, so we still have to wrap our code into a `try/catch` block. When Express 5.x will be realased, we could simplified the code. See [this](https://expressjs.com/en/guide/migrating-5.html) for more info.

So, instead of writing this in a controller : 

```ts
 catch (err) {
    return res
      .status(500)
      .json({ msg: 'E_SERVER_ERROR', stack: JSON.stringify(err) });
  }
```

we just have to write this now : 

```ts
catch (err) {
    next(err);
  }
```

## Introduce Logger

For the scalability of our platforme and the debug when the app will be in prod, we introduced a logger for our app that will keep trace of every error in `.log` files. All the errors will be kept in the `api/log/error.log` and all the other logs will be kept in the `api/log/all.log`. 

In development mode, you can print error to the console using :

```ts

import Logger from "../log/logger";

Logger.warn("this is a warning");
Logger.error("this is an error");

```

**Now, every `console.log` statement need to be rreplaced by `Logger.*` function. An eslint rules has been added to prevent using console.log`**

### CODE REVIEW

File: api/server/server.ts [hunk 1]
```diff
@@ -104,12 +105,29 @@ type ErrorOpenApi =
...
+    if (
+      (err as ErrorOpenApi).message &&
+      (err as ErrorOpenApi).errors &&
+      (err as ErrorOpenApi).status
+    ) {
+      // The error is throw by OpenApiValidator
+      Logger.info(err);
+
+      res.status((err as ErrorOpenApi).status || 500).json({
+        msg: (err as ErrorOpenApi).message,
```
Comment (alexandrelam) [inline api/server/server.ts:123]: Can you keep variable as message and not msg as we use it in front

File: api/log/logger.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  debug: 4,
+};
+
+const level = () => {
+  const env = process.env.NODE_ENV || 'development';
+  const isDevelopment = env === 'development';
+  return isDevelopment ? 'debug' : 'warn';
+};
+
+const colors = {
```
Comment (alexandrelam) [inline api/log/logger.ts:18]: Shouldn't this be uppercased ?

**Review by alexandrelam — CHANGES_REQUESTED** [review]

File: api/server/server.ts (hunk 1 above)
Comment (RomainGuarinoni) [reply api/server/server.ts:123]: Done

File: api/log/logger.ts (hunk 2 above)
Comment (RomainGuarinoni) [reply api/log/logger.ts:18]: Yes, you're right !

Comment (RomainGuarinoni) [reply api/log/logger.ts]: I"ve also add `as const` after the object declaration to make it `readonly`

**Review by AlexisAoun — APPROVED** [review]
sgtm

**Review by alexandrelam — APPROVED** [review]

**Pull request merged**